examples/ensemble/plot_clustering_on_random_trees_embedding_outputs_with_modified_distance_metrics.py

```diff
@@ -0,0 +1,177 @@
+"""
+=========================================================================
+Clustering on RandomTreesEmbedding Outputs with Modified Distance Metrics
+=========================================================================
+This demonstration is inspired by the Sparse Projection Oblique
+Randomer Forest(SPORF) package. 
+
+Distance is an important concept in clustering and other machine
+learning tasks that gives us an intuitive sense of the similarity
+between samples. This example shows how can we generate different 
+distance metrics from the tree structure info of 
+Random Tree Embedding and use these distance metrics to complete 
+various tasks. In this example, we will use them to do clustering. 
+
+The distances between two samples i and j can be defined as: 
+1) the depth of the nearest common ancestor of the leaf nodes
+sample i and j land into (take the reciprocal to make it a 
+distance); 
+2) the length of the shortest path of the leaf nodes
+sample i and j land into; 
+3) the probability of sample i and j landing into the same
+lead node(take 1 minus the value to make it a distance). 
+
+The example will be implemented on the Iris dataset as a
+simple demonstration. And the performance will be evaluated
+using ARI. 
+
+"""
+
+from sklearn import tree
+from sklearn import datasets
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.ensemble import RandomTreesEmbedding
+
+# use the Iris dataset as a simple example
+
+iris = datasets.load_iris()
```

Comment:
I would like to see this on more than just iris, lets do some simple simulation examples? How about the one from here for a start? https://scikit-learn.org/stable/auto_examples/ensemble/plot_random_forest_embedding.html#sphx-glr-auto-examples-ensemble-plot-random-forest-embedding-py

Or other examples from sklearn or jovo's papers would be a good place to start. If you have a reason to do a simulation other than those that you think would be more informative, let us know and we can discuss

Reply:
I'm thinking that this will be only an example of how to implement the code, and write a saparate  jupyter notebook that do the demonstration. I have one here but I don't know where I should put it and the notebook needs some change. Do you have any idea?

Reply:
> I'm thinking that this will be only an example of how to implement the code, and write a saparate jupyter notebook that do the demonstration. 

That's fine

> I have one here but I don't know where I should put it and the notebook needs some change. Do you have any idea?

You mean you don't know where to put it in sklearn? I wouldn't worry about that much for now, just put it wherever you put this file and we can figure that out later. Was that your question?

Reply:
I changed the notebook according to the feedback today and now it's uploaded.